source/AI.cpp

```diff
@@ -3313,32 +3313,94 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 		// Board the nearest disabled ship, focusing on hostiles before allies. Holding
 		// `Shift` results in boarding only player-owned escorts in need of assistance.
 		shared_ptr<const Ship> target = ship.GetTargetShip();
-		if(!target || !CanBoard(ship, *target) || (shift && !target->IsYours()))
+		if(!target || (target && activeCommands.Has(Command::WAIT)) || !CanBoard(ship, *target) || (shift && !target->IsYours()))
 		{
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 
-			double closest = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
-			bool foundAnything = false;
-			for(const shared_ptr<Ship> &other : ships)
-				if(CanBoard(ship, *other))
+
+			auto strategy = [&]() -> function<double(const Ship &)> {
+				Point current = ship.Position();
+				// If "board target" is set to true, the default behavior (distance priority) is used.
+				if(!Preferences::Has("Board target"))
 				{
-					if(shift && !other->IsYours())
-						continue;
+					double agility = ship.Acceleration() * ship.TurnRate();
+					return [agility, this, &ship, current](const Ship &other) -> double {
+						double cost = this->Has(ship, other.shared_from_this(), ShipEvent::SCAN_OUTFITS) ?
+							other.Cost() : (other.ChassisCost() * 2.);
+						return -agility * 2. * (cost * cost) / current.DistanceSquared(other.Position());
```

Comment:
You can remove this math.

Reply:
I suggest this logic-based approach instead of the math.  https://github.com/Hurleveur/endless-sky/pull/3

Reply:
This can be resolved.